dxp/how_clay_arrives_to_dxp.md

```diff
@@ -0,0 +1,223 @@
+
+We always recommend to use [liferay-js-themes-toolkit](https://github.com/liferay/liferay-js-themes-toolkit) to create or update a theme, but we understand that sometimes you could have the need of adding Clay to an old heavy-customized theme that can't be automatically updated. In those cases we will have to import Clay as a depencency.
+
+The first step will be to localize the project's folder and install the right dependencies. You can either use the classic Node system or the modern Yarn one.
+
+Node
+`npm install @clayui/css`
+
+Yarn
+`yarn add @clayui/css`
+
+It's important to know that these commands will only install the CSS library, if you are building a theme that can make a good use of Clay components created in React, we recommend to use:
+
+Node
+`npm install @clayui/*`
+
+Yarn
+`yarn add @clayui/*`
+
+You can also address specific components with commands like `npm install @clayui/buttons`, `yarn add @clayui/cards`, `yarn add @clayui/tables@^version`, etc. [Read more at Clay's website](https://clayui.com/docs/get-started/how-to-use-clay.html)
+
+## How is Clay imported into a module?
+
+Unlike the previous section we won't need to install Clay as a dependency in a module inside Liferay Portal. Thanks to our rich environment, we already have it included and we just need to enable it.
+
+To do so, we need to define the new import into our dedicated buildCSS process, we can do that copy the following code into the `build.gradle` file of our module.
+
+```js
+buildCSS {
+	imports = [
+		new File(npmInstall.nodeModulesDir, "@clayui/css/src/scss")
+	]
+}
+```
+
+[Real example of buildCSS](https://github.com/liferay/liferay-portal/blob/master/modules/apps/adaptive-media/adaptive-media-web/build.gradle)
+
+## How to import Clay into a custom-module?
+
+Before unreavel the final piece of this first part, we must understand what's a module and what's the difference with a theme.
+
+In Liferay we refer to a module when we talk about an indipendent bundle that can be install in DXP. It usually contains configuration, markup and style. We often call them Widgets, Portlets or Themes.
+
+So as you probably already know by this moment, there is not (technical) difference between a custom-module and a custom-theme, and as for the custom-theme we can install Clay using
+
+Node
+`npm install @clayui/css`
+
+Yarn
+`yarn add @clayui/css`
+
+[Read more at Clay's website](https://clayui.com/docs/get-started/how-to-use-clay.html)
+
+## How to use Clay once imported?
+
+And finally we arrive at the section where we explain what to do with Clay once imported. In order to have Clay available in our CSS we'll need to import it in our `main.scss`, `style.scss` or `your-custom-name-css-file.scss`.
+
+It's a really simple code line:
+
+```scss
+@import 'atlas-variables';
+```
+
+Except, _it's not_ 😁.
+
+---
+
+There are 4 different ways to import Clay CSS.
+
+```scss
+@import 'base';
+@import 'base-variables';
+@import 'atlas';
+@import 'atlas-variables';
+```
```

Comment:
I think we discussed this... the `base` and `atlas` imports have severe consequences that we consider harmful. Thus, we shouldn't document it or we should, but flagging them as "DON'T DO THIS".

In any case, we should explain what "importing Clay" means. It means those imports will end up in your compiled `.css` and duplicate selectors and everything else that's not variables or mixins. That's why this is not a good idea and we probably shouldn't even make it an option.

We're also somehow mixing how this works with guidelines for the teams. Being this a how it works, I'd expect just to document where in our classic/styled theme are we bringing clay and how. Where in our html are we loading `clay.css` and things like that.